src/main/java/com/minecolonies/core/network/messages/server/colony/ColonyFlagChangeMessage.java

```diff
@@ -44,6 +52,18 @@ public ColonyFlagChangeMessage(IColony colony, ListTag patternList)
     protected void onExecute(NetworkEvent.Context ctxIn, boolean isLogicalServer, IColony colony)
     {
         colony.setColonyFlag(patterns);
+        for (ICitizenData citizen : colony.getCitizenManager().getCitizens())
```

Comment:
In the eye of "separation of concern", this is the wrong place to put this. This should realistically live inside `colony.setFlag` and preferably should be more dynamic over just doing "instanceof JobKnight"